types/providers/index.d.ts

```diff
@@ -0,0 +1,206 @@
+// Minimum TypeScript Version: 3.8
+
+interface Providers {
+  Email: Email;
+  Credentials: Credentials;
+  Apple: Apple;
+  Twitter: Twitter;
+  Facebook: Facebook;
+  GitHub: GitHub;
+  Slack: Slack;
+  Google: Google;
+  Auth0: Auth0;
+  IdentityServer4: IdentityServer4;
+  Discord: Discord;
+  Twitch: Twitch;
+  Mixer: Mixer;
+  Okta: Okta;
+}
+
+type PossibleProviders =
+  | Email
+  | Credentials
+  | Apple
+  | Twitter
+  | Facebook
+  | GitHub
+  | Slack
+  | Google
+  | Auth0
+  | IdentityServer4
+  | Discord
+  | Twitch
+  | Mixer
+  | Okta;
+
+declare const Providers: Providers;
+export default Providers;
+export type { PossibleProviders };
+
+/**
+ * Email
+ */
+type Email = (options: ProviderEmailOptions) => void;
+
+interface ProviderEmailOptions {
+  server: string | ProviderEmailServer;
+  from: string;
+}
+
+interface ProviderEmailServer {
+  host: string;
+  port: string;
+  auth: ProviderEmailAuth;
+}
+
+interface ProviderEmailAuth {
+  user: string;
+  pass: string;
+}
+
+/**
+ * Credentials
+ */
+type Credentials = (options: ProviderCredentialsOptions) => void;
+
+interface ProviderCredentialsOptions {
+  authorizes(credentials: ProviderCredentialsObject): Promise<ProviderCredentialsObject>;
+}
+
+interface ProviderCredentialsObject {
+  [name: string]: unknown;
+}
+
+/**
+ * Apple
+ */
+type Apple = (options: ProviderAppleOptions) => void;
+
+interface ProviderAppleOptions {
+  clientId: string;
+  clientSecret: ProviderAppleSecret;
+}
+
+interface ProviderAppleSecret {
+  appleId: string;
+  teamId: string;
+  privateKey: string;
+  keyId: string;
+}
+
+/**
+ * Twitter
+ */
+type Twitter = (options: ProviderTwitterOptions) => void;
+
+interface ProviderTwitterOptions {
+  clientId: string;
+  clientSecret: string;
+}
+
+/**
+ * Facebook
+ */
+type Facebook = (options: ProviderFacebookOptions) => void;
+
+interface ProviderFacebookOptions {
+  clientId: string;
+  clientSecret: string;
+}
+
+/**
+ * GitHub
+ */
+type GitHub = (options: ProviderGitHubOptions) => void;
+
+interface ProviderGitHubOptions {
+  clientId: string;
+  clientSecret: string;
+}
+
+/**
+ * Slack
+ */
+type Slack = (options: ProviderSlackOptions) => void;
+
+interface ProviderSlackOptions {
+  clientId: string;
+  clientSecret: string;
+}
+
+/**
+ * Google
+ */
+type Google = (options: ProviderGoogleOptions) => void;
+
+interface ProviderGoogleOptions {
+  clientId: string;
+  clientSecret: string;
+}
+
+/**
+ * Auth0
+ */
+type Auth0 = (options: ProviderAuth0Options) => void;
+
+interface ProviderAuth0Options {
+  clientId: string;
+  clientSecret: string;
+  subdomain: string;
```

Comment:
From the [docs example](https://next-auth.js.org/providers/auth0#example) this field should be named `domain`

Reply:
This was already updated in the actual types PR 👍🏽